packages/react-environment/src/WebbProvider.tsx

```diff
@@ -106,20 +113,24 @@ export const WebbProvider: FC<WebbProviderProps> = ({ applicationName = 'Webb Da
         });
         const web3Provider = await Web3Provider.fromWalletConnectProvider(provider);
         await web3Provider.eth.net.isListening();
-        const net = await web3Provider.netowrk;
-        const chainType = WebbWeb3Provider.chainType(net); //  use this to pick the storage
-        const rainkybeS = await Storage.newFresh(WebbEVMChain.Rinkeby, rankebyStorage);
-        const mainS = await Storage.newFresh(WebbEVMChain.Main, mainStorage);
+        const net = await web3Provider.network; // storage based on network id
+        const chainType = WebbWeb3Provider.storageName(net);
         let storage: Storage<EVMStorage>;
-        switch (chainType) {
-          case WebbEVMChain.Main:
-            storage = mainS;
-            break;
-          case WebbEVMChain.Rinkeby:
-            storage = rainkybeS;
-            break;
-        }
-        const webbWeb3Provider = await WebbWeb3Provider.init(web3Provider, rainkybeS);
```

Comment:
It's good we init the storages lazily, however, they are needed if the user decided to switch the network from the extension, then there should be a way to swap the storage

Reply:
@nepoche  a solution for this would be to extend the EventBus for `Web3Provider` and emit an event `netowrkChanged` and pass the instance for web3Provider and then handle swapping the storage

Reply:
What about instead we refresh the application upon extension network switch as the metamask docs suggest? Then the storage would be handled by the `WebbProvider` init. I believe page refresh is how most DApps that I have used work.